### ISSUE TITLE
Make this a PR to Neovim core

### ISSUE THREAD

**Issue by mrjones2014** [nathom/filetype.nvim#36]
If this is really this much better and works consistently and as well as `filetype.vim` then IMO it should be included in Neovim core, to outright replace `filetype.vim`. Just a suggestion.

**Comment by clason** [comment]
Core is well aware of this plugin, and pretty much of the same opinion: worthwhile, but should be done carefully to avoid regressions. The more people test it now, the better :) 

Friction points to moving it upstream (which can all be overcome):
1. It's quite likely that a significant number of tests would need to be adapted.
2. It'll make it (slightly) harder to port Vim patches adding new filetypes or changing the detection.
3. The main benefit here is not Lua but the single autocommand -- would be good to quantify this (and ideally, if Vimscript is enough, make a PR to Vim).
4. To get the full benefit, you want this as a native Lua autocommand (i.e., without wrapping it in `:lua ...`), which is a non-trivial PR that is currently blocked due to lack of time.

**Comment by clason** [comment]
In the meantime, it would be useful for this plugin to stay (reasonably) up-to-date with the [`filetype.vim` updates from upstream](https://github.com/neovim/neovim/commits/master/runtime/filetype.vim) -- I'm sure these would be welcome and straightforward PRs for anybody interested in this project.

**Comment by syphar** [comment]
I believe we can close this issue, since [neovim added `filetype.lua`](https://github.com/neovim/neovim/pull/16600), see also [the discussion on reddit](https://www.reddit.com/r/neovim/comments/rvwsl3/introducing_filetypelua_and_a_call_for_help/)

**Comment by ttys3** [comment]
merged https://github.com/neovim/neovim/pull/16600

**Comment by clason** [comment]
Note that the built-in Lua filetype detection behaves somewhat differently (for the above-mentioned reasons), so you might still prefer to use this plugin.

**Comment by ahillio** [comment]
Wow cool... so will that be included in the `0.6.2` release?

**Comment by clason** [comment]
No. Features are not backported to release branches, so this will only be in 0.7.0.

**Comment by ahillio** [comment]
Thanks for clarifying :)

**Comment by clason** [comment]
Update: `filetype.lua` is now the default on Neovim master (and 0.8.0, when that is released). It has full parity with `filetype.vim` (including `scripts.vim` and `setf FALLBACK` support), passing Vim's filetype detection test suite. It will be kept in sync with future filetype detection changes in Vim and should thus be a safe and non-breaking replacement.

I believe this issue can be closed now.

**Comment by hinell** [comment]
Please close the issue, the patch has landed into `Neovim` already. See:
* [Reddit post (Jan 4, 2022)](https://www.reddit.com/r/neovim/comments/rvwsl3/introducing_filetypelua_and_a_call_for_help/)
* https://github.com/neovim/neovim/pull/16600
